bind/gencffi_type.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2017 The go-python Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package bind
+
+import (
+	"fmt"
+	"go/types"
+)
+
+// Generate Go type with Python wrapper class.
+func (g *cffiGen) genType(sym *symbol) {
+	if !sym.isType() {
+		return
+	}
+	if sym.isStruct() {
+		return
+	}
+	if sym.isBasic() && !sym.isNamed() {
+		return
+	}
+
+	pkgname := g.pkg.pkg.Name()
+	g.wrapper.Printf(`
+# Python type for %[1]s.%[2]s
+class %[2]s(object):
+    ""%[3]q""
+`, pkgname,
+		sym.goname,
+		sym.doc,
+	)
+	g.wrapper.Indent()
+	g.genTypeInit(sym)
+	g.genTypeMethod(sym)
+	g.genTypeTPStr(sym)
+	if sym.isSlice() {
+		g.genTypeIAdd(sym)
+	}
+	g.wrapper.Outdent()
+}
+
+// Generate Type __iadd__.
+func (g *cffiGen) genTypeIAdd(sym *symbol) {
+	g.wrapper.Printf("def __iadd__(self, value):\n")
+	g.wrapper.Indent()
+	switch {
+	case sym.isSlice():
+		g.wrapper.Printf("if not isinstance(value, collections.Sequence):\n")
```

Comment:
I don't think that's the test we want.

IIRC, python's `__iadd__` tests whether the value is iterable and complains if not:

```
$> python2
Python 2.7.13 (default, Jul 21 2017, 03:24:34) 
[GCC 7.1.1 20170630] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> lst=[]
>>> lst+=[1,2]
>>> lst
[1, 2]
>>> lst+=1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'int' object is not iterable
```

so I would test for `collections.Iterable` instead.
(and you actually do use the iteration protocol later on with `for elt in value:`)